[WIP] Updating Multi-Device

GrayHatter opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/utox/utox/832)
<!-- Reviewable:end -->

GrayHatter commented on src/devices.c:
```diff
@@ -1,6 +1,16 @@
 
 #include "debug.h"
+#include "friend.h" // string_to_id needs to be moved -_-
+#include "main.h"
```
this can be removed

endoffile78 commented:
Reviewed 9 of 9 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/832)*
<!-- Sent from Reviewable.io -->

robinlinden commented:
Review status: 6 of 9 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[src/av/video.c, line 308 at r2](https://reviewable.io:443/reviews/utox/utox/832#-Khwi194sPPnb6Mn8FVf:-Khwi194sPPnb6Mn8FVg:bwnnbt9) ([raw file](https://github.com/utox/utox/blob/22e2959a1ba9711ba1905008c86c52a1bfe237d4/src/av/video.c#L308)):*
> ```C
> 
>             pthread_mutex_unlock(&video_thread_lock);
>             yieldcpu(200); /* 60fps = 16.666ms || 25 fps = 40ms || the data quality is SO much better at 25... */
> ```

Sure about this? 5 fps seems a bit low.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/832)*
<!-- Sent from Reviewable.io -->

GrayHatter commented:
Review status: 6 of 9 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[src/av/video.c, line 308 at r2](https://reviewable.io:443/reviews/utox/utox/832#-Khwi194sPPnb6Mn8FVf:-Khx51G9xvngbRHKc0fM:bazjjww) ([raw file](https://github.com/utox/utox/blob/22e2959a1ba9711ba1905008c86c52a1bfe237d4/src/av/video.c#L308)):*
<details><summary><i>Previously, robinlinden (Robin Lindén) wrote…</i></summary><blockquote>

Sure about this? 5 fps seems a bit low.
</blockquote></details>

never happened :D

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/832)*
<!-- Sent from Reviewable.io -->

endoffile78 commented:
Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/832)*
<!-- Sent from Reviewable.io -->

robinlinden commented:
Reviewed 4 of 9 files at r1, 2 of 3 files at r2, 1 of 1 files at r3, 3 of 3 files at r4.
Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[src/devices.c, line 127 at r4](https://reviewable.io:443/reviews/utox/utox/832#-KiKUFm6XWm6JgHrMd_Z:-KiKUFm6XWm6JgHrMd__:b-b7bbdk) ([raw file](https://github.com/utox/utox/blob/4a30df3b3246c4206a73d665d0f53b2c4caedf48/src/devices.c#L127)):*
> ```C
>     uint16_t i;
>     for (i = 0; i < self.device_list_count; ++i) {
>         EDIT   *edit = calloc(1, sizeof(EDIT));
> ```

Change wasn't made in this PR, but I really don't like `EDIT   (lots of spaces)   *`. Now I'll have to use regex when finding all pointers to EDIT. :<

---

*[src/devices.c, line 130 at r4](https://reviewable.io:443/reviews/utox/utox/832#-KiKU2Is5IcX0v9m8a0K:-KiKU2Is5IcX0v9m8a0L:b-i2x5fe) ([raw file](https://github.com/utox/utox/blob/4a30df3b3246c4206a73d665d0f53b2c4caedf48/src/devices.c#L130)):*
> ```C
>         BUTTON *del  = calloc(1, sizeof(BUTTON));
> 
>         if (!edit) {
> ```

`!edit || !del`

---

*[src/devices.c, line 179 at r4](https://reviewable.io:443/reviews/utox/utox/832#-KiKTpEtu9hv41Wkvn-K:-KiKTpEtu9hv41Wkvn-L:bz6f8ad) ([raw file](https://github.com/utox/utox/blob/4a30df3b3246c4206a73d665d0f53b2c4caedf48/src/devices.c#L179)):*
> ```C
> 
>     if (!length_cleaned) {
>         addfriend_status = 4; // ADDF_NONAME == 4 in dns.h which I'm not willing to include;
> ```

Should the ADDF_ stuff be moved to somewhere else? addfriend_status definitely should be moved. 

---

*[src/devices.c, line 185 at r4](https://reviewable.io:443/reviews/utox/utox/832#-KiKTg9LSLpQ754-yQR2:-KiKTg9LSLpQ754-yQR3:b-vjbt2u) ([raw file](https://github.com/utox/utox/blob/4a30df3b3246c4206a73d665d0f53b2c4caedf48/src/devices.c#L185)):*
> ```C
>     uint8_t id[TOX_ADDRESS_SIZE];
> 
>     if (length_cleaned == TOX_ADDRESS_SIZE * 2 && string_to_id(id, (char*)name_cleaned)) {
> ```

Nit: Should be `char *`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/832)*
<!-- Sent from Reviewable.io -->

endoffile78 commented:
Reviewed 3 of 3 files at r4.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/832)*
<!-- Sent from Reviewable.io -->

endoffile78 commented:
Reviewed 9 of 9 files at r5.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/832)*
<!-- Sent from Reviewable.io -->

robinlinden commented:
Reviewed 8 of 9 files at r5.
Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[src/devices.c, line 59 at r5](https://reviewable.io:443/reviews/utox/utox/832#-Kiu7LziM6KPdAV2_xDj:-Kiu7LziM6KPdAV2_xDk:b6lepy7) ([raw file](https://github.com/utox/utox/blob/6c21911b04f4466fca498c7b3836b649340cc5e4/src/devices.c#L59)):*
> ```C
> };
> 
> void devices_add_new(char *name, uint16_t length, uint8_t *addr) {
> ```

`const char *` and `const uint8_t *`

---

*[src/devices.c, line 168 at r5](https://reviewable.io:443/reviews/utox/utox/832#-Kiu7hoXpTSYq8L_vqfS:-Kiu7hoXpTSYq8L_vqfT:bhy74bb) ([raw file](https://github.com/utox/utox/blob/6c21911b04f4466fca498c7b3836b649340cc5e4/src/devices.c#L168)):*
> ```C
>         edit->maxlength = TOX_PUBLIC_KEY_SIZE * 2;
>         edit->data = devices[i].pubkey_hex;
>         edit->readonly = 1;
> ```

Nit: Prefer true/false when dealing with bools.

---

*[src/friend.h, line 162 at r5](https://reviewable.io:443/reviews/utox/utox/832#-Kiu6hoU21CsWf17S4xZ:-Kiu6hoV648_yd-go2vy:b-dqyaug) ([raw file](https://github.com/utox/utox/blob/6c21911b04f4466fca498c7b3836b649340cc5e4/src/friend.h#L162)):*
> ```objc
>  */
> void binid_to_hex(char *dest, uint8_t *src);
> #define cid_to_string binid_to_hex
> ```

I don't like this. Can't you just find-and-replace them or something? Or is there a reason you did it this way?

---

*[src/cocoa/Info.plist, line 30 at r5](https://reviewable.io:443/reviews/utox/utox/832#-Kiu8K2BWxdLpYXcTtOf:-Kiu8K2BWxdLpYXcTtOg:b66860r) ([raw file](https://github.com/utox/utox/blob/6c21911b04f4466fca498c7b3836b649340cc5e4/src/cocoa/Info.plist#L30)):*
> ```XML
> 	<string>$(MACOSX_DEPLOYMENT_TARGET)</string>
> 	<key>NSHumanReadableCopyright</key>
> 	<string>Copyright © 2015-2017 Tox project. All rights reserved.</string>
> ```

These changes shouldn't've been committed, but it'll be fixed next time you merge develop into this PR, so no change required.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/832)*
<!-- Sent from Reviewable.io -->

robinlinden commented:
Reviewed 1 of 9 files at r5.
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[src/devices.c, line 204 at r5](https://reviewable.io:443/reviews/utox/utox/832#-KjExznZoI3VL3_iY2gU:-KjExznZoI3VL3_iY2gV:bnwkvld) ([raw file](https://github.com/utox/utox/blob/6c21911b04f4466fca498c7b3836b649340cc5e4/src/devices.c#L204)):*
> ```C
> 
>     if (!length_cleaned) {
>         addfriend_status = 4; // ADDF_NONAME == 4 in dns.h which I'm not willing to include;
> ```

https://reviewable.io/reviews/utox/utox/896#-KjELjfmo40SWLc1Yxn4

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/832)*
<!-- Sent from Reviewable.io -->

GrayHatter commented:
Review status: all files reviewed at latest revision, 10 unresolved discussions, some commit checks failed.

---

*[src/devices.c, line 127 at r4](https://reviewable.io:443/reviews/utox/utox/832#-KiKUFm6XWm6JgHrMd_Z:-Kjl_T1mbUddgLljuNGM:b-265ikb) ([raw file](https://github.com/utox/utox/blob/4a30df3b3246c4206a73d665d0f53b2c4caedf48/src/devices.c#L127)):*
<details><summary><i>Previously, robinlinden (Robin Lindén) wrote…</i></summary><blockquote>

Change wasn't made in this PR, but I really don't like `EDIT   (lots of spaces)   *`. Now I'll have to use regex when finding all pointers to EDIT. :<
</blockquote></details>

fixed now

---

*[src/devices.c, line 130 at r4](https://reviewable.io:443/reviews/utox/utox/832#-KiKU2Is5IcX0v9m8a0K:-Kjl_Wcg2zq77sY8zU4Y:b-896fix) ([raw file](https://github.com/utox/utox/blob/4a30df3b3246c4206a73d665d0f53b2c4caedf48/src/devices.c#L130)):*
<details><summary><i>Previously, robinlinden (Robin Lindén) wrote…</i></summary><blockquote>

`!edit || !del`
</blockquote></details>

Done.

---

*[src/devices.c, line 179 at r4](https://reviewable.io:443/reviews/utox/utox/832#-KiKTpEtu9hv41Wkvn-K:-Kjl_qNuXr_1z9cu8U0M:b-svj2c5) ([raw file](https://github.com/utox/utox/blob/4a30df3b3246c4206a73d665d0f53b2c4caedf48/src/devices.c#L179)):*
<details><summary><i>Previously, robinlinden (Robin Lindén) wrote…</i></summary><blockquote>

Should the ADDF_ stuff be moved to somewhere else? addfriend_status definitely should be moved. 
</blockquote></details>

no, it should be replaced with something better

---

*[src/devices.c, line 185 at r4](https://reviewable.io:443/reviews/utox/utox/832#-KiKTg9LSLpQ754-yQR2:-Kjl_lTw9LO6uLfiWU5Z:b-896fix) ([raw file](https://github.com/utox/utox/blob/4a30df3b3246c4206a73d665d0f53b2c4caedf48/src/devices.c#L185)):*
<details><summary><i>Previously, robinlinden (Robin Lindén) wrote…</i></summary><blockquote>

Nit: Should be `char *`
</blockquote></details>

Done.

---

*[src/devices.c, line 59 at r5](https://reviewable.io:443/reviews/utox/utox/832#-Kiu7LziM6KPdAV2_xDj:-Kjl_OvHBqau1FXuMXlJ:b-896fix) ([raw file](https://github.com/utox/utox/blob/6c21911b04f4466fca498c7b3836b649340cc5e4/src/devices.c#L59)):*
<details><summary><i>Previously, robinlinden (Robin Lindén) wrote…</i></summary><blockquote>

`const char *` and `const uint8_t *`
</blockquote></details>

Done.

---

*[src/devices.c, line 168 at r5](https://reviewable.io:443/reviews/utox/utox/832#-Kiu7hoXpTSYq8L_vqfS:-Kjl_cDRMoKV1vTozt4D:b-896fix) ([raw file](https://github.com/utox/utox/blob/6c21911b04f4466fca498c7b3836b649340cc5e4/src/devices.c#L168)):*
<details><summary><i>Previously, robinlinden (Robin Lindén) wrote…</i></summary><blockquote>

Nit: Prefer true/false when dealing with bools.
</blockquote></details>

Done.

---

*[src/devices.c, line 204 at r5](https://reviewable.io:443/reviews/utox/utox/832#-KjExznZoI3VL3_iY2gU:-Kjl_tbFVKNdeZrzZTKE:bf0jkwm) ([raw file](https://github.com/utox/utox/blob/6c21911b04f4466fca498c7b3836b649340cc5e4/src/devices.c#L204)):*
<details><summary><i>Previously, robinlinden (Robin Lindén) wrote…</i></summary><blockquote>

https://reviewable.io/reviews/utox/utox/896#-KjELjfmo40SWLc1Yxn4
</blockquote></details>

I don't wanna block this for that...

---

*[src/friend.h, line 162 at r5](https://reviewable.io:443/reviews/utox/utox/832#-Kiu6hoU21CsWf17S4xZ:-Kjla47LZGeYo1OduFtM:bei64h2) ([raw file](https://github.com/utox/utox/blob/6c21911b04f4466fca498c7b3836b649340cc5e4/src/friend.h#L162)):*
<details><summary><i>Previously, robinlinden (Robin Lindén) wrote…</i></summary><blockquote>

I don't like this. Can't you just find-and-replace them or something? Or is there a reason you did it this way?
</blockquote></details>

because I don't want to touch EVERY file for this merge.

Open an issue and assign me, I'll open a new pull for it.

Added a comment in the meantime

---

*[src/cocoa/Info.plist, line 30 at r5](https://reviewable.io:443/reviews/utox/utox/832#-Kiu8K2BWxdLpYXcTtOf:-KjlaQ8_kPexbRq9oyZN:b-896fix) ([raw file](https://github.com/utox/utox/blob/6c21911b04f4466fca498c7b3836b649340cc5e4/src/cocoa/Info.plist#L30)):*
<details><summary><i>Previously, robinlinden (Robin Lindén) wrote…</i></summary><blockquote>

These changes shouldn't've been committed, but it'll be fixed next time you merge develop into this PR, so no change required.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/832)*
<!-- Sent from Reviewable.io -->

endoffile78 commented:
Reviewed 8 of 8 files at r6.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/832)*
<!-- Sent from Reviewable.io -->
